experiments/run-experiment-01.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Runs Experiment 01 -  Optimize for incremental building
+#
+set -e
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" > /dev/null && pwd )"
+project_dir="$( pwd )"
+project_name=$( basename ${project_dir} )
+experiment_dir="$project_dir/build/enterprise-trial-experiments/experiment-01"
+
+run_id=$(uuidgen)
+
+main() {
+ info "Experiment 1: Optimize for Incremental Building"
+ info "Experiment Run ID: ${run_id}"
+ info
+
+ collect_gradle_task
+ make_experiment_dir
+ clone_project
+ execute_first_build
+ execute_second_build
+ open_build_scan_comparison
+
+ cd ${project_dir}
+ printf "\n\033[00;32mDONE\033[0m\n"
+}
+
+collect_gradle_task() {
+  read -p "What Gradle task do you want to run? (build) " task
+  if [[ "${task}" == "" ]]; then
+    task=build
+  fi
+
+  echo
+  echo
+}
+
+make_experiment_dir() {
+  mkdir -p "${experiment_dir}"
+}
+
+clone_project() {
+   info "Creating a clean clone of the project."
+
+   local clone_dir="${experiment_dir}/${project_name}"
+
+   rm -rf "${clone_dir}"
+   git clone "${project_dir}" "${clone_dir}"
```

Comment:
`git clone --depth=1` will reduce the file copying required here

Reply:
Interestingly, when I tried to use `--depth=1`, git printed out this warning:

```
warning: --depth is ignored in local clones; use file:// instead.
```

I now recall that git already does some pretty nifty things to optimize disk usage when cloning a local repository. Since commits are immutable, the original git repo, and the clone share many of the same files. I have a vague memory that it's actually using hard links to do this, but don't quote me on that.

Reply:
After the first demo (last Thursday) we decided to clone the repo from the original remote, which once again provided value to using `--depth=1`. So this has been incorporated.